tests/langchain/test_langchain_model_export.py

```diff
@@ -380,6 +382,111 @@ def load_retriever(persist_directory):
     )
 
 
+# Define a special embedding for testing
+class DeterministicDummyEmbeddings(Embeddings, BaseModel):
+    size: int
+
+    def _get_embedding(self, text: str) -> List[float]:
+        seed = abs(hash(text)) % (10**8)
+        np.random.seed(seed)
+        return list(np.random.normal(size=self.size))
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        return [self._get_embedding(t) for t in texts]
+
+    def embed_query(self, text: str) -> List[float]:
+        return self._get_embedding(text)
+
+
+def test_log_and_load_retriever_wrapper(tmp_path):
+    # Create the vector db, persist the db to a local fs folder
+    loader = TextLoader("tests/langchain/state_of_the_union.txt")
+    documents = loader.load()
+    text_splitter = CharacterTextSplitter(chunk_size=10, chunk_overlap=0)
+    docs = text_splitter.split_documents(documents)
+    embeddings = DeterministicDummyEmbeddings(size=5)
+    db = FAISS.from_documents(docs, embeddings)
+    persist_dir = str(tmp_path / "faiss_index")
+    db.save_local(persist_dir)
+
+    # Create the RetrieverWrapper
+    retriever_wrapper = RetrieverWrapper(retriever=db.as_retriever())
+
+    # Log the RetrieverWrapper
+    def load_retriever(persist_directory):
```

Comment:
Does this function need to be define here? Can we pull it out from `test_log_and_load_retriever_wrapper`?

Reply:
No, if moving it out, we will get the error here https://github.com/mlflow/mlflow/pull/8736#discussion_r1258703232.